src/vs/workbench/parts/files/common/explorerModel.ts

```diff
@@ -70,6 +70,9 @@ export class FileStat implements IFileStat {
 	public hasChildren: boolean;
 	public children: FileStat[];
 	public parent: FileStat;
+	public isVirtualDirectory: boolean;
+	public isVirtualDirectoryMember: boolean;
+	public virtualDirectoryName: string;
```

Comment:
Does it respond to creation of a new file from vscode?

Reply:
The response experience I would expect upon creating a new file, would only occur after a name has been given. Just the same as if one was to rename a file into a nested qualifying name.